core/rawdb/rawdb.go

```diff
@@ -0,0 +1,63 @@
+package rawdb
+
+import (
+	"encoding/json"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/types"
+	"github.com/ChainSafe/gossamer/polkadb"
+	log "github.com/ChainSafe/log15"
+)
+
+// check checks to see if there an error if so writes err + message to terminal
+func check(e error, msg string) {
+	if e != nil {
+		log.Crit(msg, "err", e)
+	}
+}
+
+// SetHeader stores a block header into the KV-store; key is headerPrefix + hash
+func SetHeader(db polkadb.Writer, header *types.BlockHeader) {
+	hash := header.Hash
+
+	// Write the encoded header
+	bh, err := json.Marshal(header)
+	check(err, "Failed to encode header to bytes")
```

Comment:
So the only problem with this approach is that you can't return when something fails, so the `db.Put` on the next line still happens. I'm also starting to think returning errors would make more sense than logging errors here. Another alternative is that we could panic

Reply:
We should probably panic if we are not writing to the DB properly